src/commands/edge_app.rs

```diff
@@ -391,6 +391,28 @@ impl EdgeAppCommand {
                 address_shared.lock().unwrap().as_ref().unwrap()
             );
 
+            #[cfg(target_os = "macos")]
```

Comment:
The whole code within the block_on block looks super lowbrow. What do you think about refactoring it like this:

1. Use CondVar to wait for the address. Wait with a timeout, raising an error if the app doesn't start in time. https://doc.rust-lang.org/stable/std/sync/struct.Condvar.html
2. Once the shared address is available, lock and clone it immediately so you don't need to do it later. 
3. You're not analyzing the result of the command. Do it. Right after you get the output, check that it's successful, and if not - print the error.